tests/integration_test.rs

```diff
@@ -569,3 +570,56 @@ async fn test_manifest_path_dir(#[with(PathBuf::from("examples/simple-python"))]
     assert!(pack_result.is_ok(), "{:?}", pack_result);
     assert!(pack_file.is_file());
 }
+
+#[rstest]
+#[tokio::test]
+async fn test_package_caching(
+    #[with(PathBuf::from("examples/simple-python/pixi.toml"))] options: Options,
+) {
+    let temp_cache = tempdir().expect("Couldn't create a temp cache dir");
+    let cache_dir = temp_cache.path().to_path_buf();
+
+    // First pack with cache - should download packages
+    let mut pack_options = options.pack_options.clone();
+    pack_options.cache_dir = Some(cache_dir.clone());
+    let pack_result = pixi_pack::pack(pack_options).await;
+    assert!(pack_result.is_ok(), "{:?}", pack_result);
+
+    // Get file count in cache after first pack
+    let cache_files_count = WalkDir::new(&cache_dir)
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| e.file_type().is_file())
+        .count();
+    assert!(
+        cache_files_count > 0,
+        "Cache should contain downloaded files"
+    );
+
+    // Second pack with same cache - should use cached packages
+    let temp_dir2 = tempdir().expect("Couldn't create second temp dir");
+    let mut pack_options2 = options.pack_options.clone();
+    pack_options2.cache_dir = Some(cache_dir.clone());
+    let output_file2 = temp_dir2.path().join("environment.tar");
+    pack_options2.output_file = output_file2.clone();
+
+    let pack_result2 = pixi_pack::pack(pack_options2).await;
+    assert!(pack_result2.is_ok(), "{:?}", pack_result2);
+
+    // Verify cache files weren't downloaded again by checking modification times
+    let cache_files: Vec<_> = WalkDir::new(&cache_dir)
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| e.file_type().is_file())
+        .collect();
+
+    assert_eq!(
+        cache_files.len(),
+        cache_files_count,
+        "Cache file count should remain the same"
+    );
```

Comment:
this doesn't check the modification times as you state in the comment 🤔

Reply:
Duh, that was my comment of what I wanted to do, but then I struggled to deal with that file data and defaulted to that. Great catch!

Reply:
Fixed!